src/content/posts/2023-07-12-astro-redirects.md

```diff
@@ -0,0 +1,156 @@
+
+I decided to start in the file that I would be redirecting from. For this post, I'll create a redirect page.
+
+```
+- src
+ - pages
+  - redirect
+   index.astro
+```
+
+In here I'll add a `redirect` variable. I want to drive some more traffic to our Pets of RIMdev page so I'll set that url to the value of the redirect. The frontmatter looks like this:
+
+```
+const redirect = "/the-pets-of-rimdev"
+```
+
+Now that we have our redirect defined we need to get that info to the `Head.astro` component. We pass the parameters to the layout component we are using. In this case, it's our `MainLayout.astro`.
+
+```html
+<MainLayout title={title} image={image} redirect={redirect}>
+  <h1>Redirects!</h1>
+</MainLayout>
+```
+
+Let's head to the `MainLayout` and add `redirect` to our props.
+
+```js
+const { title, image, authorImage, author, imageCredit, imageUrl, url, redirect } = Astro.props;
+```
+
+Our `Head.astro` component is used inside the `MainLayout` file so we need to pass the redirect prop there as well.
+
+```html
+<Head title={title} image={image} permalink={url} redirect={redirect} />
+```
+
+Inside our `Head.astro` component we want to add the meta refresh tag. 
+
+```html
+<!-- redirects -->
+{redirect ? (
+  <meta 
+    http-equiv="refresh" 
+    content={ `0; url=${ redirect }` } 
+  />
+) : null}
+```
+
+Here we check for `redirect` because not every page will have one so there's no point to render this. The redirect is used in the `content` which tells the browser what page to redirect to. Check it out [here](/redirect). Pretty cool. 
+
+This definitely works but I feel like we could make it a little better. We can by using [Content Collections](https://docs.astro.build/en/guides/content-collections/) and [Dynamic Pages](https://docs.astro.build/en/core-concepts/routing/#example-dynamic-pages-at-multiple-levels).
+
+## [Using Dynamic Pages](#using-dynamic-pages)
+
+We already have a dynamic page, `[slug].astro`, created for our posts. This allows us to create specific slugs for our posts. We'll use this for our redirects because we'll want to specify slugs for these redirects. Our dynamic page lives right in our `pages` folder. 
+
+```
+src
+ - content
+  - authors
+  - posts
+  - redirects
+ - pages
+  [slug].astro
+```
+
+Using Content Collections, we have a content folder inside our src directory. This is where we need to create a redirects folder. Inside the redirects folder we can add our files or pages for each redirect. Each file will represent an individual redirect. Because we're using the dynamic `[slug]` page, we can name these .md files anything we want. This will help other devs know what the file is _doing_. So we can name our redirect, `gotopets.md`. 
```

Comment:
```suggestion
Using Content Collections, we have a content folder inside our src directory. This is where we need to create a redirects folder. We can add our files or pages for each redirect in the redirects folder. Each file will represent an individual redirect. Because we're using the dynamic `[slug]` page, we can name these .md files anything we want. This will help other devs know what the file is _doing_. So we can name our redirect, `gotopets.md`. 
```